extensions/azurecore/src/azureResource/commands.ts

```diff
@@ -18,8 +18,47 @@ import { AzureResourceTreeProvider } from './tree/treeProvider';
 import { AzureResourceAccountTreeNode } from './tree/accountTreeNode';
 import { IAzureResourceSubscriptionService, IAzureResourceSubscriptionFilterService } from '../azureResource/interfaces';
 import { AzureResourceServiceNames } from './constants';
+import { AzureResourceGroupService } from './providers/resourceGroup/resourceGroupService';
 
 export function registerAzureResourceCommands(appContext: AppContext, tree: AzureResourceTreeProvider): void {
+
+	// Resource Management commands
+	appContext.apiWrapper.registerCommand('azure.accounts.getSubscriptions', async (account: azdata.Account) => {
+		const subscriptions = <azureResource.AzureResourceSubscription[]>[];
+		try {
+			const subscriptionService = appContext.getService<IAzureResourceSubscriptionService>(AzureResourceServiceNames.subscriptionService);
+			const tokens = await appContext.apiWrapper.getSecurityToken(account, azdata.AzureResource.ResourceManagement);
+
+			for (const tenant of account.properties.tenants) {
+				const token = tokens[tenant.id].token;
+				const tokenType = tokens[tenant.id].tokenType;
+
+				subscriptions.push(...await subscriptionService.getSubscriptions(account, new TokenCredentials(token, tokenType)));
+			}
+		} catch (error) {
+			throw new AzureResourceCredentialError(localize('azure.accounts.getSubscriptions.credentialError', "Failed to get credential for account {0}. Please refresh the account.", this.account.key.accountId), error);
```

Comment:
Could getSubscriptions call fail for reason other than 'failing to get credentials'?

Reply:
Yes, I need to do some work around error handling. Let me think about it and see whether I want to do that in a follow up PR or not.

Reply:
For now I made the message more generic. I'll be testing this over the next few days and will follow up with any changes I feel are necessary.

Reply:
As you think about the error scenarios, it would be great if it is possible to add tests around them.